### PR TITLE
Move PageCacheRule to IO module out of Kernel

### PR DESCRIPTION
**Pull request by MishaDemianenko**
Move PageCacheRule into io module to be accessible for all users that does not require kernel module.
Split existing PageCacheRule into base PageCacheRule that is moved to io module and ConfigurablePageCacheRule that allow additional configuration by supplier configuration.

### CODE REVIEW

File: community/common/src/test/java/org/neo4j/test/rule/ExternalResource.java [hunk 1]
```diff
@@ -24,7 +24,7 @@
 import org.junit.runners.model.Statement;
 
 /**
- * A better version of {@link org.junit.rules.ExternalResource} that properly handles exceptions in {@link #after()}.
+ * A better version of {@link org.junit.rules.ExternalResource} that properly handles exceptions in {@link #after(boolean)}.
```
Comment (burqen) [inline community/common/src/test/java/org/neo4j/test/rule/ExternalResource.java]: Line break

File: tools/src/main/java/org/neo4j/tools/dump/DumpCountsStore.java [hunk 2]
```diff
@@ -96,8 +96,8 @@ public static void dumpCountsStore( FileSystemAbstraction fs, File path, PrintSt
 
     DumpCountsStore( PrintStream out )
     {
-        this( out, Collections.<Token>emptyList(), Collections.<RelationshipTypeToken>emptyList(),
-                Collections.<Token>emptyList() );
+        this( out, Collections.emptyList(), Collections.emptyList(),
+                Collections.emptyList() );
```
Comment (burqen) [inline tools/src/main/java/org/neo4j/tools/dump/DumpCountsStore.java]: To much line break

File: community/io/src/main/java/org/neo4j/io/pagecache/impl/muninn/StandalonePageCacheFactory.java [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+/*
+ * This class is an helper to allow to construct properly a page cache in the few places we need it without all
+ * the graph database stuff, e.g., various store dump programs.
+ *
+ * All other places where a "proper" page cache is available, e.g. in store migration, should have that one injected.
+ * And tests should use the PageCacheRule.
+ */
+public final class StandalonePageCacheFactory
+{
+
```
Comment (burqen) [inline community/io/src/main/java/org/neo4j/io/pagecache/impl/muninn/StandalonePageCacheFactory.java:37]: The private constructor for `StandalonePageCacheFactory` was lost when `ConfigurableStandalonePageCacheFactory` needed to extended it. I think it should be added again.

**Review by burqen — CHANGES_REQUESTED** [review]
Looks good! Some small comments.
... well other than the build failures... that are possible to recreate locally

**Comment by burqen** [comment]
@MishaDemianenko Ready to merge once green.

**Review by burqen — APPROVED** [review]

**Comment by MishaDemianenko** [comment]
@burqen there was a problem with incorrect cache page size in newly extracted rule and that was causing new rule to be so much slower then default one. That explains why some of the tests have problems with new rule but not with old one. Now that should be fixed and all of them should work.
That change is part of last commit feel free to take a look :)

**Comment by burqen** [comment]
@MishaDemianenko Fix looks good. There is a compilation failure in `ha-robustness` though.

**Comment by MishaDemianenko** [comment]
@burqen lets merge it then and i will take a look into ha-robustness to make it work then?

**Pull request merged**